python/example_code/kms/kms_scenario.py

```diff
@@ -0,0 +1,361 @@
+        """
+        )
+        account_id = q.ask(
+            "Enter an account_id, or press enter to skip creating a grant... "
+        )
+        grant = None
+        if account_id != "":
+            grant = self.grant_manager.create_grant(
+                self.key_id,
+                account_id,
+                [
+                    "Encrypt",
+                    "Decrypt",
+                    "DescribeKey",
+                ],
+            )
+            print(f"Grant created successfully with ID: {grant['GrantId']}")
+
+        q.ask("Press Enter to continue...")
+        print(DASHES)
+        print(DASHES)
+        print(f"8. List grants for the KMS key")
+        q.ask("Press Enter to continue...")
+        self.grant_manager.list_grants(self.key_id)
+        q.ask("Press Enter to continue...")
+        print(DASHES)
+        print(f"9. Revoke the grant")
+        print(
+            """
+The revocation of a grant immediately removes the permissions and access that the grant had provided.
+This means that any account_id (user, role, or service) that was granted access to perform specific
+KMS operations on a KMS key will no longer be able to perform those operations.
+        """
+        )
+        q.ask("Press Enter to continue...")
+
+        if grant is not None:
+            self.grant_manager.revoke_grant(self.key_id, grant["GrantId"])
+            print(f"Grant ID: {grant['GrantId']} was successfully revoked!")
+
+        q.ask("Press Enter to continue...")
+        print(DASHES)
+        print(f"10. Decrypt the data\n")
+        print(
+            """
+Lets decrypt the data that was encrypted in an early step.
+The code uses the same key to decrypt the string that we encrypted earlier in the program.
+        """
+        )
+        q.ask("Press Enter to continue...")
+        decrypted_data = self.key_encryption.decrypt(self.key_id, encrypted_text)
+        print(f"Data decrypted successfully for key ID: {self.key_id}")
+        print(f"Decrypted data: {decrypted_data}")
+
+        q.ask("Press Enter to continue...")
+        print(DASHES)
+        print(f"11. Replace a key policy\n")
+        print(
+            """
+A key policy is a resource policy for a KMS key. Key policies are the primary way to control
+access to KMS keys. Every KMS key must have exactly one key policy. The statements in the key policy
+determine who has permission to use the KMS key and how they can use it.
+You can also use IAM policies and grants to control access to the KMS key, but every KMS key
+must have a key policy.
+
+By default, when you create a key by using the SDK, a policy is created that
+gives the AWS account that owns the KMS key full access to the KMS key.
+
+Let's try to replace the automatically created policy with the following policy.
+{
+"Version": "2012-10-17",
+"Statement": [{
+"Effect": "Allow",
+"Principal": {"AWS": "arn:aws:iam::0000000000:root"},
+"Action": "kms:*",
+"Resource": "*"
+}]
+}
```

Comment:
This has the potential to get out of sync with the actual policy.

Reply:
I agree. On the other hand, I want to keep the follow as close as possible to the scout. This was in the scout. I am going to leave it as is.